src/chainstate/stacks/db/transactions.rs

```diff
@@ -411,7 +422,14 @@ impl StacksChainState {
                     Error::ClarityError(e)
                 })?;
 
-                Ok(asset_map.get_stx_burned_total())
+                let receipt = StacksTransactionReceipt {
```

Comment:
Do we want to preserve the error reason in `result`?

Reply:
Hmm github might be off on your comment, not sure to understand.
![Screenshot from 2020-04-03 13-51-05](https://user-images.githubusercontent.com/87777/78390324-5b067600-75b2-11ea-941f-219d1e3408f5.png)

Reply:
If an error is thrown, we're not returning the receipt (unwrapped by the `?` above).

Reply:
I see, thank you for unpacking. I'm personally in favor of a dedicated cause/diagnostic field tracking legible messages (which is not something we have at this point).
I think that the way we handle runtime errors deserve some love. Can we address this in a separate PR?

Reply:
Sure, it can be a separate PR.  Can you create an issue for it?

Reply:
Issue available here https://github.com/blockstack/stacks-blockchain/issues/1392